### PR TITLE
[IMP] mail: support oEmbed link preview

### PR DESCRIPTION
**Pull request by phenix-factory**
oEmbed (https://oembed.com) is a format for allowing an embedded representation
of a URL on third party sites like Odoo.

task-2365881

### CODE REVIEW

**Comment by xmo-odoo** [comment]
> One solution, would be to create a white list of providers that we trust, and only display the html for those providers.
> What do you think ?

From the official specification:

> Consumers may wish to load the HTML in an off-domain iframe to avoid XSS vulnerabilities.

**Comment by mart-e** [comment]
> One solution, would be to create a white list of providers that we trust, and only display the html for those providers.

But do we really trust some providers? I definitly don't want to embed JS from Twitter or alikes.
Even if we did trust them, the point of the standard is to be usable on any website following it. Getting a different render on some arbitrary websites doesn't seem like the way to go.
As mentionned Xavier, an isolated iframe seems safer if you want to allow unsanitzed HTML (but not convinced that you should).

File: addons/mail/security/ir.model.access.csv [hunk 1]
```diff
@@ -1,5 +1,6 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_mail_message_all,mail.message.all,model_mail_message,,1,0,0,0
+access_mail_message_oembed,mail.message.oembed,model_mail_oembed,,1,0,0,0
```
Comment (mart-e) [inline addons/mail/security/ir.model.access.csv:3]: Anyway can read the embed from anyone? That should be linked to the visibility of the attachment

Comment (phenix-factory) [reply addons/mail/security/ir.model.access.csv]: All right. Doesn't change much for me.

**Comment by xmo-odoo** [comment]
> As mentionned Xavier, an isolated iframe seems safer if you want to allow unsanitzed HTML (but not convinced that you should).

FWIW the sanitiser does not strip *just* scripts, the original comment mentions iframes which I can see being an issue: I would expect the richer embeds to just embed their own content in a frame (not unlike the way you'd embed bits of odoo) but this gets stripped by the standard san.

And the sanitiser doesn't allow customising the tags whitelist.

**Comment by phenix-factory** [comment]
@xmo-odoo This mean we have to have a service like https://oembed.link/ or https://iframely.com/. I guess it could be done for Saas people. What about the others ?

@mart-e I guess we never want to trust anyone by default.
To be fair, I would be happy to sanitize the html but just keep the iframe. But I can't do that, and it doesn't cover every case of oembed data.

**Pull request closed without merging**